### PR TITLE
convince mypy that user.is_staff (and friends) are booleans

### LINKED ISSUE #512: user.is_staff complex type
Hello, as I am not sure if this is an issue related to django-stubs, or djangorestframework-stubs, or just my code, I am posting here as well. I created this issue https://github.com/typeddjango/djangorestframework-stubs/issues/109
Seems `user.is_staff` does not agree with the `bool` return type annotation I gave to my method. Is this an issue for fixing in django-stubs or djangorestframework-stubs? If it is an issue of my code, please suggest what I could do.

### PR DESCRIPTION
**Pull request by ghost**
# i have made things!

## related issues

closes #512

this should be enough to convince mypy that the `is_staff`, `is_active`, and `is_superuser` are (for all intents and purposes) bools

alternatively we could set things up so that `BooleanField`s are *always* seen by mypy as a bool, but i think that a conservative approach is best

### CODE REVIEW

File: mypy_django_plugin/main.py [hunk 1]
```diff
@@ -270,8 +270,12 @@ def get_attribute_hook(self, fullname: str) -> Optional[Callable[[AttributeConte
             return partial(settings.get_type_of_settings_attribute, django_context=self.django_context)
 
         info = self._get_typeinfo_or_none(class_name)
+        if info and info.has_base(fullnames.PERMISSION_MIXIN_CLASS_FULLNAME) and attr_name == "is_superuser":
+            return partial(set_auth_user_model_boolean_fields, django_context=self.django_context)
         if info and info.has_base(fullnames.HTTPREQUEST_CLASS_FULLNAME) and attr_name == "user":
             return partial(request.set_auth_user_model_as_type_for_request_user, django_context=self.django_context)
+        if info and info.has_base(fullnames.ABSTRACT_USER_MODEL_FULLNAME) and attr_name in ("is_staff", "is_active"):
```
Comment (sobolevn) [inline mypy_django_plugin/main.py:277]: I am a big fan of class-level immutable constants for cases like this. Do you think that this is worth refactoring?

**Review by sobolevn — COMMENTED** [review]
Awesome! Thanks!

File: mypy_django_plugin/main.py (hunk 1 above)
Comment (sobolevn) [reply mypy_django_plugin/main.py:277]: `frozenset` basically 🙂

Comment (ghost) [reply mypy_django_plugin/main.py:277]: do you mean moving `ABSTRACT_USER_MODEL_FULLNAME` to the `NewSemanalDjangoPlugin` class?

Comment (sobolevn) [reply mypy_django_plugin/main.py:277]: No, I was thinking about `('is_staff', 'is_active')`

Comment (ghost) [reply mypy_django_plugin/main.py:277]: ahhh
yeah, tbh i was trying to think of a way to turn it into a dict, so you could just pass in the fullname and attr_name and get the correct partial as a result, but i'm not sure that's doable in a clean way :/

i don't think that one bit is worth refactoring, but i absolutely want to refactor those three if statements

Comment (sobolevn) [reply mypy_django_plugin/main.py:277]: Ok 👍

**Review by sobolevn — APPROVED** [review]

**Pull request merged**